src/Exthost/Extension/Contributions.re

```diff
@@ -371,6 +374,9 @@ module Language = {
         {
           id: field.required("id", string),
           extensions: field.withDefault("extensions", [], list(string)),
+          filenames: field.withDefault("filenames", [], list(string)),
```

Comment:
I want to look up what vscode does when loading these. The Ruby file defines `gemfile` as the filename, but most projects seem to use `Gemfile` (i.e. capitalised) so a straight look up won't work. Then the shell one defines them in all caps.

Should we just lower case/upper case all file names and do the same to the filenames? Or is there a reason to not do that?

Reply:
Thanks for looking at this, @CrossR !

> The Ruby file defines gemfile as the filename, but most projects seem to use Gemfile (i.e. capitalised) so a straight look up won't work. 

Good catch.

> Should we just lower case/upper case all file names on loading and do the same to the filenames? Or is there a reason to not do that?

I don't think we want to change the actual filename / path - as some filesystems are case sensitive, so it would change the file location. But we should do a case-insensitive compare when we check these fields - we could normalize both the 'filenames' entry and the file we check to lowercase-ascii when we check to see if this filenames field matches.